Dockerfile

```diff
@@ -1,37 +1,37 @@
-# Settings.
+# Build arguments.
 ARG USER_ID=3001
 ARG USER_NAME=vireo
-ARG SOURCE_DIR=/$USER_NAME/source
+ARG HOME_DIR=/$USER_NAME
+ARG SOURCE_DIR=$HOME_DIR/source
+ARG APP_PATH=/var/vireo
 ARG NODE_ENV=production
 
 # Maven stage.
-FROM maven:3-openjdk-11-slim as maven
+FROM maven:3-eclipse-temurin-11-alpine as maven
 ARG USER_ID
 ARG USER_NAME
+ARG HOME_DIR
 ARG SOURCE_DIR
+ARG APP_PATH
 ARG NODE_ENV
 
 ENV NODE_ENV=$NODE_ENV
 
-# Create the user and group (use a high ID to attempt to avoid conflicts).
-RUN groupadd --non-unique -g $USER_ID $USER_NAME && \
-    useradd --non-unique -d /$USER_NAME -m -u $USER_ID -g $USER_ID $USER_NAME
+# Create the group (use a high ID to attempt to avoid conflits).
+RUN addgroup -g $USER_ID $USER_NAME
 
-# Install stable Nodejs and npm.
-RUN apt-get update && \
-    apt-get upgrade -y && \
-    apt-get install -y nodejs npm iproute2 && \
-    apt-get clean && \
-    rm -rf /var/lib/apt/lists/* && \
-    npm cache clean -f && \
-    npm install -g n && \
-    n stable
+# Create the user (use a high ID to attempt to avoid conflits).
+RUN adduser -h $HOME_DIR -u $USER_ID -G $USER_NAME -D $USER_NAME
 
 # Ensure source directory exists and has appropriate file permissions.
 RUN mkdir -p $SOURCE_DIR && \
     chown $USER_ID:$USER_ID $SOURCE_DIR && \
     chmod g+s $SOURCE_DIR
 
+# Upgrade the system and install dependencies.
+RUN apk -U upgrade && \
+    apk add --update --no-cache nodejs npm make g++ py3-pip
```

Comment:
I thought all of that cache clean code in the old apt-get update was necessary.
Is `apk` different enough that this is not a problem?
Or perhaps, it just doesn't support that functionality?

Reply:
Pending the base image. The updates over time become a vulnerability.

Reply:
I am asking about `apk` usage here.
Do you know that `apk` does not need to cache changes that `apt-get` does?